### PR TITLE
feat: Add registration status for sale card

### PR DESCRIPTION
**Pull request by jo-rs**
The type of this PR is: Feature

This PR resolves [PURCHASE-2167]

### Description

User will see the following CTA, if they're not registered which will redirect them to register, or if their identity needs to be verified which will redirect them to the faq.
Screenshot 2020-12-02 at 19 16 39

And the following CTA if their registration is pending.
Screenshot 2020-12-02 at 19 17 42

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[PURCHASE-2167]: https://artsyproduct.atlassian.net/browse/PURCHASE-2167

### CODE REVIEW

**Review by erikdstock — COMMENTED** [review]
This looks great from here - wondering if we can add any tests for it? I'd be willing to pair when i'm up tomorrow.

**Comment by erikdstock** [comment]
Actually, 2 more (non-blocking) thoughts here:
1 - It is actually possible to link directly to the identity verification flow as we have done in force. However it may not be trivial. The FAQ might be a fine implementation for now, but I'm not sure it will tell the user they need to check their email, as I think it is even more general than that.
2 - I think we should have an empty, qualified for bidding sale card as well. This could be a separate PR, but this card would link to the auction screen so user can easily get back to a sale they have expressed interest in. However from looking at the designs I see we haven't called that out specifically. Something to consider but non-blocking.

**Comment by jo-rs** [comment]
> This looks great from here - wondering if we can add any tests for it? I'd be willing to pair when i'm up tomorrow.

Sure thing. I'll start putting some in, and will ping you to check that/pair if needed. Thanks, Erik!

**Comment by jo-rs** [comment]
> Actually, 2 more (non-blocking) thoughts here:
> 1 - It is actually possible to link directly to the identity verification flow as we have done in force. However it may not be trivial. The FAQ might be a fine implementation for now, but I'm not sure it will tell the user they need to check their email, as I think it is even more general than that.
> 2 - I think we should have an empty, qualified for bidding sale card as well. This could be a separate PR, but this card would link to the auction screen so user can easily get back to a sale they have expressed interest in. However from looking at the designs I see we haven't called that out specifically. Something to consider but non-blocking.

1- I followed what was already implemented in the app, but would be happy to put in the other stuff. Might need some help if you're up for it, though.
2- Should be easy to implement once there's a design. Whatever needs to be rendered can just be dropped in [here](https://github.com/artsy/eigen/blob/1a240b64b2c47ca269afb7d6dd879acfafdbb643/src/lib/Scenes/MyBids/Components/SaleCard.tsx#L49)

**Review by erikdstock — APPROVED** [review]
Looks good, merging!

File: src/lib/Scenes/MyBids/Components/SaleCard.tsx [hunk 1]
```diff
@@ -1,17 +1,67 @@
...
+import { SaleCard_me } from "__generated__/SaleCard_me.graphql"
 import { SaleCard_sale } from "__generated__/SaleCard_sale.graphql"
 
 export const CARD_HEIGHT = 72
 
-export class SaleCard extends React.Component<{ sale: SaleCard_sale; smallScreen?: boolean }> {
+export const RegistrationCTAWrapper: React.FunctionComponent<{ navLink?: string }> = (props) => (
+  <Touchable
+    underlayColor={props.navLink ? "black5" : "transparent"}
+    onPress={() => props.navLink && navigate(props.navLink)}
```
[the remark this answers is not in the document]
Comment (jo-rs) [reply src/lib/Scenes/MyBids/Components/SaleCard.tsx:18]: Sorry, I missed that comment of Friday 😞

**Pull request merged**